File: Count the subarrays having product less than k[4-July-2023]/solution.cpp
class Solution{
  public:
    int countSubArrayProductLessThanK(const vector<int>& a, int n, long long k) {
        int i=0,j=0;
        long prod=1;
        int ans=0;
        while(j<n)
        {
            prod*=a[j];
            
            while(i<=j && prod>=k)
            {
                prod/=a[i];
                i++;
            }
            
            ans+=(j-i+1);
            j++;
        }
        
            return ans;
    }
};

File: Count the subarrays having product less than k[4-July-2023]/solution.py
// >> PYTHON CODE <<

class Solution:
    def countSubArrayProductLessThanK(self, a, n, k):
        
        i,j=0,0
        prod=1
        ans=0
        while j<n:
            prod*=a[j]
            
            
            while(i<=j and prod>=k):
                prod/=a[i]
                i+=1
            
            ans+=(j-i+1)
            j+=1
        
        return ans